src/ucp/core/ucp_worker.c

```diff
@@ -3012,6 +3046,12 @@ void ucp_worker_keepalive_add_ep(ucp_ep_h ep)
         return;
     }
 
+#if UCS_ENABLE_ASSERT
+    ucp_ep_ext_control(ep)->ka_count = 0;
+#endif
+    ucs_assert(worker->keepalive.num_eps < UINT_MAX);
+    ++worker->keepalive.num_eps;
```

Comment:
why need to track the number of keepalive eps? can't we just remove the progress callback when the list is empty?